src/hypothesis/searchstrategy/strings.py

```diff
@@ -133,3 +134,43 @@ def __init__(self, size):
 
     def do_draw(self, data):
         return binary_type(data.draw_bytes(self.size))
+
+
+class EmailStrategy(MappedSearchStrategy):
+    VALID_CHARACTERS = one_of(
+        sampled_from((string.ascii_letters, string.digits, "!#$%&'*+-/=?^_`{|}~")),
+        characters(min_codepoint=0x007F)
+    )
+
+    def __init__(self, domains, min_size, average_size, max_size):
+        super(EmailStrategy, self).__init__(
+            strategy=fixed_dictionaries({
+                'local-part': builds(
+                    intersperse_dots,
+                    randoms(),
+                    text(
+                        LOCAL_PART,
+                        min_size=min_size, average_size=average_size, max_size=max_size
+                    )),
+                'domain': domains
+            })
+        )
+
+
+    def pack(self, value):
+        return "%s@%s" % (
+            value['local-part'], value['domain']
+        )
+
+
+def intersperse_dots(random, string):
```

Comment:
Yeah, all of the email stuff can pretty much be ignored, as I haven't worked on it since when I started this endeavor.